valhalla/thor/triplegbuilder.h

```diff
@@ -25,22 +25,40 @@ namespace thor {
 class TripLegBuilder {
 public:
   /**
-   * Format the trip path output given the edges on the path.
-   * For now just return length. TODO - modify to return trip path.
+   * Form a trip leg out of a path (sequence of path infos)
+   *
+   * @param controller            Which meta data attributes to include in the trip leg
+   * @param graphreader           A way of accessing graph information
+   * @param mode_costing          A costing object
+   * @param path_begin            The first path info in the path
+   * @param path_end              One past the last path info in the path
+   * @param origin                The origin location with path edges filled in from loki
+   * @param dest                  The destination location with path edges filled in from loki
+   * @param through_loc           The list of through locations along this leg if any
+   * @param trip_path             The leg we will fill out
+   * @param interrupt_callback    A way to abort the processing in case the request was cancelled
+   * @param route_discontinuities Markers at places in the leg where there are discontinuities
+   * @param trim_begin            For map matching we have one long sequence of path infos regardless
+   *                              of legs so we must supply an amount of elapsed time which we trim
+   *                              from the beginning
+   * @param trim_end              Similarly to trim_begin, we must also trim at the end of a map
+   *                              matched edge
+   * @return
```

Comment:
just needed to update this for the two new params. it was missing javadoc which is a crime considering how important this method is